tests/Unit/Products/ProductImageTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Tests\Unit\Products;
+
+use Illuminate\Contracts\Filesystem\Factory as FilesystemContract;
+use Illuminate\Http\UploadedFile;
+use IndieHD\Velkart\Base\Traits\UploadsFiles;
+use IndieHD\Velkart\ProductImage\Contracts\ProductImageRepositoryContract;
+use IndieHD\Velkart\Tests\TestCase;
+
+class ProductImageTest extends TestCase
```

Comment:
We may want to break these tests up into smaller, more specific tests. It should be a simple copy/paste refactor.

I think there's merit to the idea that "a test should fail for only one reason". This is a pretty good discussion of the premise:

https://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test

I definitely have tests with more than one assertion, but when I do, both assertions are effectively testing the same thing (only because, in some rare instances, one assertion alone can't quite prove the condition).

Reply:
I can certainly break these down and i do agree. At least as much as possible 1 assert per test.